view/src/tests/DiffViewWrapper.test.jsx

```diff
@@ -321,4 +323,106 @@ describe('testing a caller function visualisation', () => {
       () => {},
     );
   });
+
+  // Integration testing from Difference component.
+  test('code of calls which are only on one side of call stack should be visualised correctly', async () => {
+    // Note: For these calls should be shown code of the function/macro and it should
+    // be shown on correct side (left/right), the other side should be left empty.
+
+    // Note: Reusing parts from previous test case.
+    const example = {
+      results: [{
+        // compared function
+        function: 'cmp_fun',
+        diffs: [{
+          // differing macro
+          function: 'diff',
+          'old-callstack': [
+            { name: 'left (macro)', line: 5, file: 'file1.c' },
+            { name: 'diff (macro)', line: oldStart, file: 'left_file.c' },
+          ],
+          'new-callstack': [
+            { name: 'right (macro)', line: 5, file: 'file1.c' },
+            { name: 'diff (macro)', line: newStart, file: 'right_file.c' },
+          ],
+        }],
+      }],
+      definitions: {
+        left: {
+          kind: 'macro',
+          old: { line: oldStart, file: 'left_file.c', 'end-line': oldEnd },
+          diff: false,
+        },
+        right: {
+          kind: 'macro',
+          new: { line: newStart, file: 'right_file.c', 'end-line': newEnd },
+          diff: false,
+        },
+      },
+    };
+
+    // Mocking getFile to return the content of the file.
+    const getFile = jest.fn(async (filePath) => {
+      let content = filePath;
+      if (filePath === 'old-src/left_file.c') {
+        content = oldCode;
+      } else if (filePath === 'new-src/right_file.c') {
+        content = newCode;
+      }
+      return content;
+    });
+
+    render(
+      <Difference
+        compare={example.results[0].function}
+        diff={example.results[0].diffs[0]}
+        definitions={example.definitions}
+        getFile={getFile}
+        oldFolder=""
+        newFolder=""
+      />,
+    );
+
+    const callstack = within(screen.getByTestId('callstack'));
+
+    // Selecting function which is only in the old call stack.
+    userEvent.click(callstack.getByTitle('left (macro)'));
+    // Waiting for update.
+    await screen.findByText(/left_file.c/);
+    // Testing that the whole code of the function is shown
+    // and it is shown on the left side and the right side is empty.
```

Comment:
The latter part ("right side is empty") is not tested, is it?

Reply:
Good question. It was actually tested by providing the empty string to `testCodeNumberMatch` function
https://github.com/PLukas2018/diffkemp/blob/8c561152923644fbfb537a1693bc0f59197eaf9e/view/src/tests/DiffViewWrapper.test.jsx#L393-L397
, which would fail here
https://github.com/PLukas2018/diffkemp/blob/8c561152923644fbfb537a1693bc0f59197eaf9e/view/src/tests/DiffViewWrapper.test.jsx#L54
if the side was not empty (if a line would be shown).

I have felt like the `testCodeNumberMatch` function was not clearly documented so I added a commit that adds comments to make it hopefully more clear, also I simplified one thing by using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#optional_chaining_with_function_calls).